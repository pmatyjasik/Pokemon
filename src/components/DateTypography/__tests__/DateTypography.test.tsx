import { render, screen } from "@testing-library/react";
import {
    DateTypography,
    DateTypographyProps,
} from "@/components/DateTypography/DateTypography";

describe("Date Component", () => {
    const renderComponent = (props: DateTypographyProps) => {
        return render(<DateTypography {...props} />);
    };

    const testValue = "2024-08-20";

    it("Matches DOM Snapshot", () => {
        const { asFragment } = render(<DateTypography date={testValue} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the date prop correctly", () => {
        renderComponent({ date: testValue });

        expect(screen.getByText(testValue)).toBeInTheDocument();
    });
});
