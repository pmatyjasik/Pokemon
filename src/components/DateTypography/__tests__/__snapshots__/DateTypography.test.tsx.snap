// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Component Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <h6
    class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-alignRight css-19ne65h-MuiTypography-root"
  >
    2024-08-20
  </h6>
</DocumentFragment>
`;
